### PR TITLE
P20-972: Update teacher students modal to support grace period

### PR DESCRIPTION
**Pull request by artem-vavilov**
## Screenshot
cap_teacher_warning_banner

## Links
- JIRA ticket: [P20-972](https://codedotorg.atlassian.net/browse/P20-972)
- Figma: [Teacher Warning Banner](https://www.figma.com/design/wR0Myx8T2dHMaeeb1GAabb/CPA-Flows?node-id=1981-283&t=bwkQLJFvZztbe2pq-0)
- Related modal PR: https://github.com/code-dot-org/code-dot-org/pull/58413
- Related "Grace Period" PR: https://github.com/code-dot-org/code-dot-org/pull/59192

### CODE REVIEW

**Review by juanmanzojr — COMMENTED** [review]
Could you add to the unit tests within `AgeGatedStudentsTableTest.tsx` testing the Parent/Guardian Emailed? column maps correctly
[Link to Test](https://github.com/code-dot-org/code-dot-org/blob/be29c5ade59104d028e589db8dec66b935b92ac0/apps/test/unit/templates/AgeGatedStudentsTableTest.tsx#L164)

**Review by daynew — APPROVED** [review]
Looks good to me besides addressing the comment Juan already raised.

File: apps/test/unit/templates/AgeGatedStudentsTableTest.tsx
Comment (juanmanzojr) [inline apps/test/unit/templates/AgeGatedStudentsTableTest.tsx:1]: Thanks for adding and refactoring!

**Review by juanmanzojr — APPROVED** [review]
LGTM!

**Pull request merged**